sdk/core/System.ClientModel/src/Pipeline/MessageLoggingPolicy.cs

```diff
@@ -79,12 +74,12 @@ private async ValueTask ProcessSyncOrAsync(PipelineMessage message, IReadOnlyLis
         string requestId = Guid.NewGuid().ToString();
         message.Request.ClientRequestId = requestId;
 
-        _handler.LogRequest(request, requestId, _clientAssembly);
+        _messageLogger.LogRequest(requestId, request, _clientAssembly);
```

Comment:
Curious: could the call to LogRequest on the logging abstraction check the options and log the request content if needed?

Reply:
This is an interesting idea. From a technical perspective, this would work. To avoid too much code duplication, it would involve also moving the response content log.
That would mean moving all of the "heavy lifting" out of the message logging policy and into the logging abstraction. I would prefer keeping the implementation in the message logging policy, since the goal of the logging abstraction is to help limit maintainability problems around having multiple logging sinks. I like having the logging abstraction simply doing log formatting and logging to either Event Source or ILogger.